src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static int DEFAULT_CAPACITY = 10;
+    private T[] arrayElements;
+    private int size;
+
+    public ArrayList() {
+        arrayElements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public T[] resize() {
+        T[] oldArrayElements = arrayElements;
+        arrayElements = (T[]) new Object[arrayElements.length + (arrayElements.length >> 1)];
+        System.arraycopy(oldArrayElements, 0, arrayElements, 0, oldArrayElements.length);
+        return arrayElements;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size > arrayElements.length - 1) {
+            resize();
+        }
+        arrayElements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("The index is not correct");
+        }
+        if (size > arrayElements.length) {
+            resize();
+        }
+        System.arraycopy(arrayElements, index, arrayElements, index + 1, size - index);
+        arrayElements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (size > arrayElements.length) {
+            resize();
+        }
```

Comment:
if we call add() for each element of the list, do we need to check capacity here?

Reply:
Done